### PR TITLE
Health potion

### PR DESCRIPTION
**Pull request by lsizemore8**
Implemented the health potion. Tested on iOS and windows. To test, play island-1. It'll appear near the end of the level on the leftmost column. In the level json, it's represented as a canvas with only [11] for colors. Swiping vertically where you start on the bottle and end outside of the bottle before the timer runs out clears the bottle & gives you some health back. 

Some points to consider: 
1. Given the danger bar is still a danger bar, the math for this works as if this will remain a danger bar. This should not be a major thing to fix, as all we need to do is switch where current_health is calculated/set. 
2. Right now, doing a horizontal swipe or double tap will make the health potion go away as "wrong". This can be easily changed but I am waiting on discussion from the team. 
3. It's only in one level right now (island_1). I'll make a second PR at add it to more later.

### CODE REVIEW

File: source/scenes/gameplay/PPGameScene.cpp
Comment (AnshG714) [inline source/scenes/gameplay/PPGameScene.cpp]: This was addressed in a small PR i made that is open rn, but this will cause merge conflicts. To account for the timed out canvases adding to the danger bar, all we need to do is change this line to 
`max((float)(_state.getScoreMetric("wrongAction") + _state.getScoreMetric("timedOut")) - _state.getHealthBack(), 0.0f);` and it'll fix the issue of timed out canvases not affecting the danger bar. Figured we'll do it on your PR to avoid conflicts.

File: source/controllers/PPGameStateController.cpp
Comment (AnshG714) [inline source/controllers/PPGameStateController.cpp:43]: would it be better to have one common matrix for obstacles and health portions, and make them int arrays instead of bool arrays? Just for the sake of using fewer data structures for state management. It could be as simple as using 1 to indicate obstacles, 2 for health potions, 3 for beach balls (if that gets integrated), and -1 for 'normal' canvases. This is just a nit, and we can probably fix that later but if you have time this may be worth thinking about?

**Review by AnshG714 — COMMENTED** [review]
This PR seems great, look at my comments! The array fix i mentioned is a low priority so feel free to ignore that. I'm just holding off on approving so you can see my comment about integrating the timed out actions into this code (look at #70). 

Health potions seem to be working fine on my end. I've tried the following:
- Swiping up dismissing the health potion and reducing danger bar
- Swiping left and right (incorrect dismissal)
- Swiping down (no effect)
- Letting the health potion be (it times out quickly)

All these are expected behaviors, so WOOHOO!

File: source/controllers/PPGameStateController.cpp
Comment (lsizemore8) [reply source/controllers/PPGameStateController.cpp:43]: I had considered that. At first it was mainly to keep logic for checking way simpler & less refactoring for obstacles but in hindsight it shouldn't be too much of a change to fix. I can try to fix that now

File: source/scenes/gameplay/PPGameScene.cpp
Comment (lsizemore8) [reply source/scenes/gameplay/PPGameScene.cpp]: Sounds good. I'll fix that now

**Review by BryanGraeser — APPROVED** [review]
Tested it on Android and it works, though it's a little finicky (it didn't work the first time I tried and because it's at the end of the level I don't want to replay it like all again to repeatedly test). I think just keep an eye on how easy/hard to is to swipe

**Pull request merged**
